extensions/rest-client/runtime/src/main/java/io/quarkus/restclient/runtime/RestClientRecorder.java

```diff
@@ -69,6 +69,7 @@ public InjectorFactory getInjectorFactory() {
         }
 
         RestClientBuilderImpl.setProviderFactory(clientProviderFactory);
+        ResteasyProviderFactory.setInstance(clientProviderFactory);
```

Comment:
This won't conflict with the server behavior when both are used? /cc @kenfinnigan

Reply:
I don't think so. When the server starts (if any) it will replace the instance:

https://github.com/resteasy/Resteasy/blob/1570f6825de1fde1ea3c3a950dcb6297e3c9bd4f/resteasy-core/src/main/java/org/jboss/resteasy/core/ResteasyDeploymentImpl.java#L377

And it will work for both server and client with the right factory.

Reply:
Is the `ResteasyDeploymentImpl` used in Quarkus, for some reason I'm thinking it isn't, but could easily be wrong?

If we do use it, then I think it should work fine. But certainly worth verifying

Reply:
Yes, it is used here:
https://github.com/quarkusio/quarkus/blob/5941a39ba4c1eecedd854f74b4d9e688082a73fc/extensions/resteasy-server-common/runtime/src/main/java/io/quarkus/resteasy/server/common/runtime/QuarkusResteasyDeployment.java

Reply:
Thanks for checking

Reply:
Famous Last Words™